perf(mangler): optimize slot assignment with pre-computed ancestor sets

camc314 opened the pull request:
<img width="701" height="488" alt="Screenshot 2025-12-26 at 22 08 31" src="https://github.com/user-attachments/assets/8061127d-c75e-47ef-b2ef-899adad590c7" />

Boshen commented:
### Merge activity

* **Dec 27, 5:45 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 27, 5:45 AM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

camc314 committed:
perf(mangler): optimize slot assignment with pre-computed ancestor sets (#17379)

<img width="701" height="488" alt="Screenshot 2025-12-26 at 22 08 31" src="https://github.com/user-attachments/assets/8061127d-c75e-47ef-b2ef-899adad590c7" />
